python/ray/serve/examples/tutorials/sklearn+composition.rst

```diff
@@ -0,0 +1,344 @@
+            if serve.context.web:
+                payload = flask_request.json
+            else:
+                payload = payload
+    
+            input_vector = [
+                payload["sepal length"],
+                payload["sepal width"],
+                payload["petal length"],
+                payload["petal width"],
+            ]
+            prediction = self.model.predict([input_vector])[0]
+            human_name = self.label_list[prediction]
+            return human_name
+
+Serve’s endpoints are logical names for the service. It can be
+associated with several backends. Each backend correspond to a function
+or a class.
+
+.. code:: ipython3
+
+    serve.create_endpoint(endpoint_name="logistic", route="/logistic")
+    serve.create_endpoint(endpoint_name="svm", route="/smv")
+
+.. code:: ipython3
+
+    serve.create_backend(SKLearnModel, "logistic:v1", "/tmp/iris_logistic_regression.pkl")
+    serve.create_backend(SKLearnModel, "svm:v1", "/tmp/iris_svm_classifier.pkl")
+
+.. code:: ipython3
+
+    serve.link("logistic", "logistic:v1")
+    serve.link("svm", "svm:v1")
+
+How can we implement the ensemble function? Each endpoints in Serve is
+reachable via RayServeHandle. You can directly call the it via python.
+
+.. code:: ipython3
+
+    handle = serve.get_handle("svm")
+    handle
+
+
+
+
+.. parsed-literal::
+
+    
+    RayServeHandle(
+        Endpoint="svm",
+        URL="http://127.0.0.1:8000/svm",
+    )
+
+
+
+.. code:: ipython3
+
+    sample_input = {
+        "sepal length": 1.2,
+        "sepal width": 1.0,
+        "petal length": 1.1,
+        "petal width": 0.9,
+    }
+
+When you invoke the handle, an ``ObjectID`` is returned immediately. The
+``ObjectID`` contains the future result from prediction. You can
+retrieve the result via ``ray.get``
+
+.. code:: ipython3
+
+    object_id = handle.remote(payload=sample_input)
+    object_id
+
+
+
+
+.. parsed-literal::
+
+    ObjectID(0bead116322a6c2b93415054010000c801000000)
+
+
+
+.. code:: ipython3
+
+    import ray
+    ray.get(object_id)
+
+
+
+
+.. parsed-literal::
+
+    'versicolor'
+
+
+
+We can implement our ensemble function this way.
+
+.. code:: ipython3
+
+    @serve.route("/ensemble")
+    def ensemble(flask_request):
+        payload = flask_request.json
+    
+        # Both requests are fired at the same time
+        logistic_out_object_id = serve.get_handle("logistic").remote(payload=payload)
+        svm_out_object_id = serve.get_handle("svm").remote(payload=payload)
+    
+        # We retrieve from both models together
+        logistic_out, svm_out = ray.get([logistic_out_object_id, svm_out_object_id])
+    
+        # We will return both output
+        return {"logistic": logistic_out, "svm": svm_out}
+
+.. code:: ipython3
+
+    print(requests.get("http://127.0.0.1:8000/ensemble", json=sample_input).text)
+
+
+.. parsed-literal::
+
+    {
+      "logistic": "virginica",
+      "svm": "versicolor"
+    }
+
+
+What just happened?
```

Comment:
I think it will be good addition if we can explain how to run this with autoscaler as some people asked about this. (It can be a followup tutorial). Also, will you write about batching example in a different tutorial?